.github/workflows/run_tests.yml

```diff
@@ -1,9 +1,35 @@
-name: Android build PR
-on: [pull_request]
+name: Run tests
+
+on:
+  push:
+    branches:
+    - '**'
+  pull_request:
+    branches:
+    - main
+
 jobs:
+  check_duplicate_runs:
```

Comment:
If someone pushes commit to a new branch, and then creates a pull request to the same repo (e.g. from feature branch into main branch), Github runs the same CI workflow twice. This job allows to skip the second run of the workflow if another workflow was started by the same commit.